pkg/base/config.go

```diff
@@ -55,7 +55,39 @@ const (
 
 	// defaultRangeLeaseRaftElectionTimeoutMultiplier specifies what multiple the
 	// leader lease active duration should be of the raft election timeout.
-	defaultRangeLeaseRaftElectionTimeoutMultiplier = 3
+	//
+	// Timers for Raft leadership election and lease expiration run in parallel.
+	// Although not required, we would like to elect a leader before the lease
+	// expires, such that we don't have to wait for a Raft election when we're
+	// ready to acquire the lease.
+	//
+	// The relevant operations and default time intervals are (assuming RTTs
+	// ranging from 10ms to 400ms):
+	//
+	// Raft election:
+	// - Heartbeat offset (0-1 heartbeat interval)   [-1.00s - 0.00s]
+	// - Election timeout (random 1x-2x timeout)     [ 2.00s - 4.00s]
+	// - Election (3x RTT: prevote, vote, append)    [ 0.03s - 1.20s]
+	//
+	// Lease acquisition:
+	// - Heartbeat offset (0-1 heartbeat interval)   [-2.50s - 0.00s]
+	// - Lease expiration (constant)                 [ 5.00s        ]
+	// - Liveness epoch bump (2x RTT: CPut + append) [ 0.02s - 0.80s]
+	// - Lease acquisition (1x RTT: append)          [ 0.01s - 0.40s]
+	//
+	// From the above, we note that the worst-case Raft election latency
+	// (4.03s-5.20s) is always less than the corresponding lease expiration +
+	// epoch bump time (5.02s-5.80s) regardless of RTT, such that the upper bound
+	// on unavailability is always given by the lease expiration time + 3x RTT
+	// (5.03s to 6.20s).
+	//
+	// With negligible RTT, the average latency is 3.75s for lease acquisition
+	// (-2.5s / 2 + 5.0s) and 2.5s for Raft elections (-1.0s / 2 + (2.0s + 4.0s) /
+	// 2). However, the worst-case Raft election latency (4.0s) being greater than
+	// the best-case lease acquisition latency (2.5s) for a given RTT will skew
+	// the average upwards, so we can approximate the typical unavailability to be
+	// roughly 4.0s (the exact calculation is left as an exercise for the reader).
+	defaultRangeLeaseRaftElectionTimeoutMultiplier = 2.5
```

Comment:
How do you feel about a little test (`echotest.Require`) that takes the default config and spits out the resulting lease/election durations, etc, so that we're able to see all of this math play out in the actual numbers that are going to be used (vs just trusting/verifying the comments)?

Reply:
Done, including the pre-change state.

Reply:
This is good, but I meant the analysis above (in the comment), which changes with any tweak to the consts, no? Optional but maybe this is enough to bait you

```go
func foo() {
	var cfg base.RaftConfig // TODO
	var lo, hi time.Duration
	var buf strings.Builder
	phase := func(pl, ph time.Duration, expl string) {
		fmt.Fprintf(&buf, "[%.2fs - %.2fs] %s\n", pl.Seconds(), ph.Seconds(), expl)
		lo += pl
		hi += ph
	}
	{
		// Last heartbeat is at best one interval in the past, at worst it just happened.
		lo := -1 * cfg.RaftTickInterval * time.Duration(cfg.RaftHeartbeatIntervalTicks)
		phase(lo, 0, "last heartbeat")
	}
	{
		lo := cfg.RaftElectionTimeout()
		var rcfg *raft.Config
		rcfg.Mul
		hi := 2 * lo // see (*raft).resetRandomizedElectionTimeout.
		phase(lo, hi, "randomized election timeout")
	}
	{
		// ...
	}
	fmt.Fprintf(&buf, "[%.2fs - %.2fs] %s\n", lo.Seconds(), hi.Seconds(), "*** TOTAL ***")
}

```

Reply:
Oh, I see. Might as well write a test to compute it, so that people don't forget to update the comment when they change the parameters.

Reply:
Done.